### DISCUSSION TITLE
systemfit and stagefit

### DISCUSSION THREAD

**Discussion by shin258** [ikemen-engine/Ikemen-GO#1871] [Feature Request]
### Describe the solution you'd like

Add the option as similar to The KeepAspect option on 03-07-2024 DEV build.

### Additional context

on mugen 1.1d4, I used the systemfit option to have the screenpack in large screen and stagefit to keep the appearance of the stages on which I play (this can be bypassed on ikemengo put 1280x960 what I do currently + The keepaspect option in 03-07-2024 DevBuild.) ![in fight](https://github.com/ikemen-engine/Ikemen-GO/assets/69587714/ffd296ce-f27d-4a81-9043-c8ca0653b93f) but because of this I have the screenpack displayed incorrectly due to the fact that I have to stay in 1280x960 (without the systemfit option i cannot switch into 1920x1080 or better)
![screenpack without the systemfit option](https://github.com/ikemen-engine/Ikemen-GO/assets/69587714/f2f34173-6393-40a7-9013-b664165280f8)
I play exclusively with levels of mugen and mugen-hi
My question is:
It is possible to add these two option at the same as the keep aspect option in the last build (03-07-2024) ?
thank you also for your work (I already switch definitely on ikemengo whatever happens)

### Search terms

systemfit stagefit

**Comment by SuperFromND** [comment]
This is something I'd like to do in the future, but I want to first improve our implementation of window-scaling (the way I implemented it is rather _hacky_ and could definitely be improved).

**Reply by shin258** [reply]
in my side the window-scaling works very well on my side because I do not use shader or msaa.

I wonder, the option is hard to integrate because ikemengo works different from the official version of mugen1.1? I made a video on mugen1.1 that shows how great the systemfit/stagefit option is for the characters and stages of the winmugen and winmugen plus version.

https://github.com/user-attachments/assets/a1ec23cd-d7c3-4415-9718-c04593336ea5

**Reply by shin258** [reply]
I have a question. Do you think also, the possibility hack or extend the off value of 4:3 stages to wild-screen by the engine? With this possibility the system and stage fit option are finaly not required.
